### PR TITLE
Add a FAQ page

### PR DESCRIPTION
**Pull request by mlvandijk**
Add a FAQ page with answers to frequently asked question.
This is a partial fix for isse #52; not all questions mentioned in the issue are answered atm.

### CODE REVIEW

File: content/community/faq.md [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+By default Cucumber-JVM will search in the package (or sub-packages) of the runner class.
+You can also tell Cucumber-JVM explicitly which packages (and sub-packages) to search, with:
+ ```java
+ @CucumberOptions(glue = {"<package>", "<package>", "<etc>"})
+ public class RunCucumberTest{}
+```
+{{% /block %}}
+
+## How do I call other steps or scenarios?
+Each scenarios should be *independent*; you should be able to run them in any order or in parallel without one scenario interfering with another.
```
Comment (msmorawski) [inline content/community/faq.md]: Typo, should be `Each scenario`

File: content/community/faq.md [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+
+Each scenario should *test exactly one thing* so that when it fails, it fails for a clear reason. This means you wouldn't reuse one scenario inside another scenario.
+
+If your scenarios use the same or similar steps, or or perform similar actions on your system, you can extract [helper methods](#how-to-use-helper-methods) to do those things.
```
Comment (msmorawski) [inline content/community/faq.md]: Repeated `or`

File: content/community/faq.md [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+  // sendKeys to password fields
+
+  // click Login button
+
+}
+```
+
+You can call this method from your step definition. You could even call *several* helper methods from one step definition; for instance, if you want to login and navigate to a particular page.
+
+Ofcourse, which helper methods you'll need will depend on your application and your domain.
```
Comment (msmorawski) [inline content/community/faq.md]: Should be `Of course`

File: content/community/faq.md [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+# How do I share state between steps?
+{{% block "ruby,javascript" %}}
+In order to share state between steps, you can use a [world object](/cucumber/state/#world-object).
+{{% /block %}}
+
+{{% block "java,kotlin" %}}
+If you are using Cucumber on the JVM, you can use [dependency injection (DI)](/cucumber/state/#dependency-injection) to share state between steps.
+
+If your project already uses a dependency framework supported by Cucumber (and/or you are familiar with one of them), it's probably easiest to use that framework. 
+Otherwise, Picocontainer is the most light weight framework you can use.
```
Comment (msmorawski) [inline content/community/faq.md:148]: Maybe should be replaced by `lightest weight`?

Comment (mlvandijk) [reply content/community/faq.md:148]: Hmm, not sure. Using "light weight" as an expression here.

Comment (mlvandijk) [reply content/community/faq.md:148]: Fixed the typos though, thanks for pointing them out!

**Pull request merged**